connect/connection.go

```diff
@@ -59,10 +60,23 @@ func successCode(code int) bool {
 	return code >= 200 && code < 300
 }
 
+func proxyWithAuth(req *http.Request) (*url.URL, error) {
+	proxyUrl, err := http.ProxyFromEnvironment(req)
+	// nil proxyUrl indicates no proxy configured
+	if proxyUrl == nil || err != nil {
+		return proxyUrl, err
+	}
+	// add or replace proxy credentials if configured
+	if c, err := readCurlrcCredentials(curlrcCredentialsFile()); err == nil {
```

Comment:
There's no real "parse error" possible. In the end we need to know if we found credentials or not. If file exists but has nothing interesting inside, it's the same as if it didn't exist for us.
Remember that `.curlrc` is not just for proxy passwords. People could have this file with any custom settings they want. If after attempted read we don't end up with credentials, there's nothing we could (or should imo) do. i.e. I don't think this is an error which should be reported to the user in any way.

Reply:
I was actually hesitant about implementing this `ErrNoProxyCredentials` error but rather checking for empty credentials or something. In the end I realized that this is exactly how golang logic of error/exception handling works. Not finding proxy credentials IS an error in scope of this function because it should "read and return the credentials", but it is NOT an exception or anything critical from user point of view.

Reply:
And the calling code looks way cleaner without all those additional checks :o)

Reply:
ah ok - .curlrc is not just for proxy passwords.